### PR TITLE
fix: sync OAuth tokens to base keyring username for load_tokens() compatibility

### LINKED ISSUE #984: OAuth tokens not synced between keyring and file storage after re-authentication
## Description

After re-authenticating via `mcporter auth mcp-atlassian --reset`, the OAuth session configuration fails with:

```
Error calling tool 'search': Failed to configure OAuth session
```

## Root Cause

The token loading logic in `OAuthConfig.load_tokens()` prioritizes keyring over file storage:

1. User re-authenticates → new tokens saved to **both** keyring and file
2. However, the keyring entry for the cloud-specific username (`oauth-{client_id}-cloud-{cloud_id}`) may not be updated correctly
3. On next MCP call, `load_tokens()` retrieves **stale tokens from keyring**
4. `ensure_valid_token()` tries to refresh but fails (403 Forbidden - refresh token already rotated)
5. `configure_oauth_session()` returns False

## Evidence

```python
# Keyring has old expires_at
>>> keyring.get_password("mcp-atlassian-oauth", "oauth-{client_id}")
{"expires_at": 1771648288, ...}  # Expired

# File has fresh tokens
>>> json.load(open("~/.mcp-atlassian/oauth-{client_id}.json"))
{"expires_at": 1771934383, ...}  # Valid
```

## Workaround

Manually sync file tokens to keyring:

```python
import keyring, json
tokens = json.load(open("~/.mcp-atlassian/oauth-{client_id}.json"))
keyring.set_password("mcp-atlassian-oauth", "oauth-{client_id}", json.dumps(tokens))
```

## Suggested Fix

In `_save_tokens()`, ensure both keyring entries are updated:
- `oauth-{client_id}` (used by `load_tokens()`)
- `oauth-{client_id}-cloud-{cloud_id}` (used by `_get_keyring_username()`)

Or consolidate to a single keyring key pattern.

## Environment

- mcp-atlassian version: 0.16.1
- Python: 3.13
- OS: macOS (Sequoia)
- Keyring backend: macOS Keychain

### PR DESCRIPTION
**Pull request by Shahid-LT**
## Summary

Fixes #984

When saving tokens, `_save_tokens()` uses `_get_keyring_username()` which includes `cloud_id` or `base_url` hash in the key (e.g., `oauth-{client_id}-cloud-{cloud_id}`). However, `load_tokens()` uses a simpler pattern (`oauth-{client_id}`) without the context suffix.

This mismatch causes `load_tokens()` to return stale tokens from keyring after re-authentication, because the new tokens are saved under a different key.

## Changes

In `_save_tokens()`, also save tokens to the base `oauth-{client_id}` keyring entry when the context-specific username differs. This ensures `load_tokens()` always finds the latest tokens.

## Testing

1. Authenticate with OAuth: `mcporter auth mcp-atlassian`
2. Wait for token to expire or manually expire it
3. Re-authenticate: `mcporter auth mcp-atlassian --reset`
4. Verify MCP calls work without "Failed to configure OAuth session" error

Before this fix, step 4 would fail because `load_tokens()` would retrieve stale tokens from the `oauth-{client_id}` keyring entry.

### CODE REVIEW

File: src/mcp_atlassian/utils/oauth.py [hunk 1]
```diff
@@ -366,11 +367,18 @@ def _save_tokens(self) -> None:
...
-            # Store the token data in the system keyring
-            keyring.set_password(KEYRING_SERVICE_NAME, username, json.dumps(token_data))
+            token_json = json.dumps(token_data)
 
+            # Store the token data in the system keyring using context-specific key
+            keyring.set_password(KEYRING_SERVICE_NAME, username, token_json)
             logger.debug(f"Saved OAuth tokens to keyring for {username}")
 
+            # Also save to base username for compatibility with load_tokens()
+            # which uses the simpler oauth-{client_id} pattern
```
Comment (sooperset) [inline src/mcp_atlassian/utils/oauth.py]: (minor) If the same `client_id` is used for both Cloud and DC (unusual but possible), the base key gets overwritten by whichever saved last. This is fine for the standard single-instance setup, but a brief inline comment noting this would help future readers.

File: tests/unit/utils/test_oauth.py [hunk 2]
```diff
@@ -355,16 +355,24 @@ def test_save_tokens_keyring_success(self, mock_save_to_file, mock_set_password)
...
-        mock_set_password.assert_called_once()
-        service_name = mock_set_password.call_args[0][0]
-        username = mock_set_password.call_args[0][1]
-        token_json = mock_set_password.call_args[0][2]
-
-        assert service_name == KEYRING_SERVICE_NAME
-        assert username == "oauth-test-client-id-cloud-test-cloud-id"
-        assert "test-refresh-token" in token_json
-        assert "test-access-token" in token_json
+        # Verify keyring was used - should be called twice:
```
Comment (sooperset) [inline tests/unit/utils/test_oauth.py:358]: (minor) This validates the dual-write for Cloud context. Consider adding a similar case for DC context to verify the `dc-{url_hash}` username also triggers the base key write.

**Review by sooperset — DISMISSED** [review]
Good catch on the keyring key mismatch. The dual-write approach is simple and brings keyring into parity with the file storage path (which already uses the base `oauth-{client_id}` pattern). Tests and lint pass cleanly.

File: src/mcp_atlassian/utils/oauth.py (hunk 1 above)
Comment (Shahid-LT) [reply src/mcp_atlassian/utils/oauth.py]: Added inline comment noting the Cloud/DC overwrite behavior for the rare edge case. Thanks for the review!

File: tests/unit/utils/test_oauth.py (hunk 2 above)
Comment (Shahid-LT) [reply tests/unit/utils/test_oauth.py:358]: Added `test_save_tokens_keyring_success_dc` to verify the dual-write works for Data Center context with the `oauth-{client_id}-dc-{url_hash}` pattern. Good suggestion!
